### ISSUE TITLE
Disabling Extensions

### ISSUE THREAD

**Issue by rizar** [bartvm/blocks#320]
I want to be able to specify all extensions I may need in the `MainLoop` `extensions` argument and then to switch of unnecessary:

```
main_loop = MainLoop(...)
main_loop.find_extension("Dump").enabled = False
```

That might be implemented by introducing `_dispatch` in addition to `dispatch` in the `TrainingExtension`, but this will decrease readability. Should be also possible to do with a metaclass, and this seems to be a better solution.

**Comment by bartvm** [comment]
Does this not work?

```python
    def _run_extensions(self, method_name, *args):
        for extension in self.extensions:
            if getattr(extension, 'enabled', True):
                extension.dispatch(method_name, *args)
```

**Comment by rizar** [comment]
Right, I did not think about that, but it can also be implemented in the main loop. Just that it seems to me a bit more logical to have this functionality implemented in the `TrainingExtension`. But simplicity overweights.
